apps/web/src/hooks/useArtworkPreview.ts

```diff
@@ -0,0 +1,223 @@
+import React, { BaseSyntheticEvent } from 'react'
+
+import { OrderedTraits } from 'src/modules/create-dao/components/Artwork/LayerBox'
+
+import { ImageProps } from './useArtworkUpload'
+
+export interface UseArtworkPreviewProps {
+  orderedLayers: OrderedTraits
+  images?: ImageProps[]
+}
+
+export interface ImagesByTraitProps {
+  trait: string
+  images: ImageProps[]
+}
+
+export interface SelectedTraitsProps {
+  picker: string
+  trait: string
+  uri: string
+  content: File
+}
+
+export const useArtworkPreview = ({ images, orderedLayers }: UseArtworkPreviewProps) => {
+  const canvas = React.useRef(null)
+  const [generatedImages, setGeneratedImages] = React.useState<any[]>([])
```

Comment:
it would be great to streamline these types around artwork upload. maybe that's worth tackling separately though as I wouldn't be surprised if that might lead to some cascading type incompatibility lol.